orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationConfig.java

```diff
@@ -1,136 +1,29 @@
-   * @see TemplateConfig
-   * @see OrchestrationPrompt
-   */
-  @Nonnull
-  Option<TemplateConfig> getTemplate();
-
-  /**
-   * The masking configuration.
-   *
-   * @return An Option containing the masking configuration if set, empty otherwise
-   * @see MaskingConfig
-   * @see DpiMaskingConfig
-   */
-  @Nonnull
-  Option<MaskingConfig> getMaskingConfig();
-
-  /**
-   * The input content filter.
-   *
-   * @return An Option containing the input filter configuration if set, empty otherwise
-   * @see #getOutputContentFilter()
-   * @see ContentFilter
-   * @see AzureContentFilter
-   */
-  @Nonnull
-  Option<ContentFilter> getInputContentFilter();
-
-  /**
-   * The output content filter.
-   *
-   * @return An Option containing the output filter configuration if set, empty otherwise
-   * @see #getInputContentFilter()
-   * @see ContentFilter
-   * @see AzureContentFilter
-   */
-  @Nonnull
-  Option<ContentFilter> getOutputContentFilter();
-
-  /**
-   * Set the language model configuration.
-   *
-   * @param llm The LLM configuration to be used
-   * @return Fluent API object of this config instance
-   * @see AiModel
-   * @see LlmConfig
-   */
-  @Nonnull
-  T withLlmConfig(@Nonnull final AiModel llm);
-
-  /**
-   * Set the template configuration for prompt composition.
-   *
-   * @param templateConfig The template configuration to be used
-   * @return Fluent API object of this config instance
-   * @see TemplateConfig
-   * @see OrchestrationPrompt
-   */
-  @Nonnull
-  T withTemplate(@Nonnull final TemplateConfig templateConfig);
-
-  /**
-   * Set a data masking configuration.
-   *
-   * @param maskingConfig The masking configuration to be used
-   * @return Fluent API object of this config instance
-   * @see MaskingConfig
-   * @see DpiMaskingConfig
-   */
-  @Nonnull
-  T withMaskingConfig(@Nonnull final MaskingConfig maskingConfig);
-
-  /**
-   * Configure an input content filter.
-   *
-   * @param filter The content filter configuration to be applied to input
-   * @return Fluent API object of this config instance
-   * @see #withOutputContentFilter(ContentFilter)
-   * @see ContentFilter
-   * @see AzureContentFilter
-   */
-  @Nonnull
-  T withInputContentFilter(@Nonnull final ContentFilter filter);
-
-  /**
-   * Configure an output content filter.
-   *
-   * @param filter The content filter configuration to be applied to output
-   * @return Fluent API object of this config instance
-   * @see #withInputContentFilter(ContentFilter)
-   * @see ContentFilter
-   * @see AzureContentFilter
-   */
-  @Nonnull
-  T withOutputContentFilter(@Nonnull final ContentFilter filter);
+@Value
+@With
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class OrchestrationConfig {
+
+  @Nullable AiModel llmConfig;
+  @Nullable TemplateConfig template;
+  @Nullable MaskingConfig maskingConfig;
+  @Nullable ContentFilter inputContentFilter;
+  @Nullable ContentFilter outputContentFilter;
+
+  public OrchestrationConfig() {
```

Comment:
Slight variation:

```suggestion
  public OrchestrationConfig(AiModel llmConfig) {
```

Upside: Users see that this is required, downside: users have to provide it initially, even if they want to change it per request.